### ISSUE TITLE
plugins/micron: Compiler warning on Fedora 38

### ISSUE THREAD

**Issue by martin-belanger** [linux-nvme/nvme-cli#1974]
I get this warning when I compile nvme-cli on Fedora 38, using gcc (GCC) 13.1.1 20230511 (Red Hat 13.1.1-2):
```
ninja -C .build
ninja: Entering directory `.build'
[48/82] Compiling C object nvme.p/plugins_micron_micron-nvme.c.o
../plugins/micron/micron-nvme.c: In function ‘micron_internal_logs’:
../plugins/micron/micron-nvme.c:3263:33: warning: argument 1 null where non-null expected [-Wnonnull]
 3263 |                                 memset(dataBuffer, 0, bSize);
      |                                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from ../plugins/micron/micron-nvme.c:3:
/usr/include/string.h:61:14: note: in a call to function ‘memset’ declared ‘nonnull’
   61 | extern void *memset (void *__s, int __c, size_t __n) __THROW __nonnull ((1));
      |              ^~~~~~
[82/82] Linking target nvme
```

**Comment by ikegami-t** [comment]
Sorry for the error. Fixed this by the PR #1975.

**Comment by ikegami-t** [comment]
I tried to confirm the build warning with the Fedora and gcc versions mentioned but the warning was not happend. Can you advice any about this?

**Comment by martin-belanger** [comment]
Hi @ikegami-t - I just confirmed that the fix https://github.com/linux-nvme/nvme-cli/pull/1975 solves the issue.

**Comment by ikegami-t** [comment]
Thanks for the confirmation. I thought to check if any other warning outputs more to fix.

**Comment by igaw** [comment]
I've merged #1975.
